src/contracts/core/PaymentCoordinator.sol

```diff
@@ -0,0 +1,474 @@
+        distributionRoots.push(
+            DistributionRoot({
+                root: root,
+                activatedAt: activatedAt,
+                paymentCalculationEndTimestamp: paymentCalculationEndTimestamp
+            })
+        );
+        currPaymentCalculationEndTimestamp = paymentCalculationEndTimestamp;
+        emit DistributionRootSubmitted(rootIndex, root, paymentCalculationEndTimestamp, activatedAt);
+    }
+
+    /**
+     * @notice Create a RangePayment. Called from both `payForRange` and `payAllForRange`
+     */
+    function _payForRange(RangePayment calldata rangePayment) internal {
+        require(rangePayment.strategiesAndMultipliers.length > 0, "PaymentCoordinator._payForRange: no strategies set");
+        require(rangePayment.amount > 0, "PaymentCoordinator._payForRange: amount cannot be 0");
+        require(
+            rangePayment.duration <= MAX_PAYMENT_DURATION,
+            "PaymentCoordinator._payForRange: duration exceeds MAX_PAYMENT_DURATION"
+        );
+        require(
+            rangePayment.duration % calculationIntervalSeconds == 0,
+            "PaymentCoordinator._payForRange: duration must be a multiple of calculationIntervalSeconds"
+        );
+        require(
+            rangePayment.startTimestamp % calculationIntervalSeconds == 0,
+            "PaymentCoordinator._payForRange: startTimestamp must be a multiple of calculationIntervalSeconds"
+        );
+        require(
+            block.timestamp - MAX_RETROACTIVE_LENGTH <= rangePayment.startTimestamp &&
+                GENESIS_PAYMENT_TIMESTAMP <= rangePayment.startTimestamp,
+            "PaymentCoordinator._payForRange: startTimestamp too far in the past"
+        );
+        require(
+            rangePayment.startTimestamp <= block.timestamp + MAX_FUTURE_LENGTH,
+            "PaymentCoordinator._payForRange: startTimestamp too far in the future"
+        );
+
+        // Require rangePayment is for whitelisted strategy or beaconChainETHStrategy
+        address currAddress = address(0);
+        for (uint256 i = 0; i < rangePayment.strategiesAndMultipliers.length; ++i) {
+            IStrategy strategy = rangePayment.strategiesAndMultipliers[i].strategy;
+            require(
+                strategyManager.strategyIsWhitelistedForDeposit(strategy) ||
+                    strategy == beaconChainETHStrategy,
+                "PaymentCoordinator._payForRange: invalid strategy considered"
+            );
+            require(
+                currAddress < address(strategy),
+                "PaymentCoordinator._payForRange: strategies must be in ascending order to handle duplicates"
+            );
+            currAddress = address(strategy);
+        }
+
+        rangePayment.token.safeTransferFrom(msg.sender, address(this), rangePayment.amount);
+    }
+
+    function _checkClaim(PaymentMerkleClaim calldata claim, DistributionRoot memory root) internal view returns (bool) {
```

Comment:
The `bool` return for this method is kinda useless, because if anything fails verification, the method will revert.

Can we remove this?

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/514/commits/24fb70726a39a125e61cc19a361d7cc20aa7b930